x-pack/plugins/enterprise_search/public/applications/shared/truncate/truncated_content.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { truncate, truncateBeginning } from './';
+
+interface ITruncatedContentProps {
+  content: string;
+  length: number;
+  beginning?: boolean;
+  tooltipType?: 'inline' | 'title';
+}
+
+export const TruncatedContent: React.FC<ITruncatedContentProps> = ({
+  content,
+  length,
+  beginning = false,
+  tooltipType = 'inline',
+}) => {
+  if (content.length <= length) return <>{content}</>;
+
+  const inline = tooltipType === 'inline';
+  return (
+    <span className="truncated-content" title={!inline ? content : ''}>
+      {beginning ? truncateBeginning(content, length) : truncate(content, length)}
+      {inline && <span className="truncated-content__tooltip">{content}</span>}
```

Comment:
Since we're using classes, should we also port styles?

Reply:
Great catch! Done